Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs

```diff
@@ -169,35 +173,98 @@ public static AuthenticationBuilder AddCustomAuthentication(this IServiceCollect
                 .AddScheme<AuthenticationSchemeOptions, CustomAuthenticationHandler>(AuthenticationSchemes.CustomAuthentication, null);
         }
 
+        /// <summary>
+        /// Sets up the proxy configuration based on the addresses in <paramref name="userList"/>.
+        /// </summary>
+        /// <param name="networkManager">The <see cref="INetworkManager"/> instance.</param>
+        /// <param name="config">The <see cref="NetworkConfiguration"/> containing the config settings.</param>
+        /// <param name="userList">The string array to parse.</param>
+        /// <param name="options">The <see cref="ForwardedHeadersOptions"/> instance.</param>
+        public static void ParseList(INetworkManager networkManager, NetworkConfiguration config, string[] userList, ForwardedHeadersOptions options)
+        {
+            for (var i = 0; i < userList.Length; i++)
+            {
+                if (IPNetAddress.TryParse(userList[i], out var addr))
+                {
+                    if ((!config.EnableIPV4 && addr.AddressFamily == AddressFamily.InterNetwork)
+                         || (!config.EnableIPV6 && addr.AddressFamily == AddressFamily.InterNetworkV6))
+                    {
+                        continue;
+                    }
+
+                    if (networkManager.SystemIP6Enabled && addr.AddressFamily == AddressFamily.InterNetwork)
+                    {
+                        // If the server is using dual-mode sockets, IPv4 addresses are supplied in an IPv6 format.
+                        // https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-5.0 .
+                        addr.Address = addr.Address.MapToIPv6();
+                    }
+
+                    if (addr.PrefixLength == 32)
+                    {
+                        options.KnownProxies.Add(addr.Address);
+                    }
+                    else
+                    {
+                        options.KnownNetworks.Add(new IPNetwork(addr.Address, addr.PrefixLength));
+                    }
+                }
+                else if (IPHost.TryParse(userList[i], out var host))
+                {
+                    foreach (var address in host.GetAddresses())
+                    {
+                        if ((!config.EnableIPV4 && host.AddressFamily == AddressFamily.InterNetwork)
+                            || (!config.EnableIPV6 && host.AddressFamily == AddressFamily.InterNetworkV6))
```

Comment:
This is not dependent on `address`, so just move it out of the loop.

Reply:
Good catch! Meant to be address. Updated